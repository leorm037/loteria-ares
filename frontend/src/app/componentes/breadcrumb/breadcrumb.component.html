<nav id="breadcrumb" aria-label="breadcrumb" class="bg-loteria-secondary shadow border-bottom border-loteria"
    style="--bs-breadcrumb-divider: '>';">
    <ol class="breadcrumb m-0 p-2">
        <li class="breadcrumb-item">
            <a routerLink="/" class="text-decoration-none">
                <i class="bi bi-house-fill"></i> PÃ¡gina inicial
            </a>
        </li>
        @for (breadcrumb of breadcrumbs; track breadcrumb.texto) {
            @if (breadcrumb.href == null) {
            <li class="breadcrumb-item active">
                @if (breadcrumb.iconClass) {
                    <i class="{{ breadcrumb.iconClass }}" aria-hidden="true"></i>  
                }
                {{ breadcrumb.texto }}
            </li>
            } @else {
                <li class="breadcrumb-item">
                    <a routerLink="{{ breadcrumb.href }}" class="text-decoration-none">
                        @if (breadcrumb.iconClass) {
                            <i class="{{ breadcrumb.iconClass }}" aria-hidden="true"></i>  
                        }
                        {{ breadcrumb.texto }}
                    </a>
                </li>
            }
        }
    </ol>
</nav>