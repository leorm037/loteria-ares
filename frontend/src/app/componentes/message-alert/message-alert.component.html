@for (message of messagesAlert; track message.message) {
    <div class="alert alert-{{ message.type }}" role="alert">
        @if (message.iconClass) {
            <i class="{{ message.iconClass }}" aria-hidden="true"></i> 
        }
        {{ message.message }}
    </div>
}