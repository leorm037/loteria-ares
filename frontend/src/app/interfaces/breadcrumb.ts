export interface Breadcrumb {
    href?: string;
    iconClass?: string;
    texto: string;
}
