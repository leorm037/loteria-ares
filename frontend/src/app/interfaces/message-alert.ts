import { MessageAlertType } from "../enum/message-alert-type";

export interface MessageAlert {
    type: MessageAlertType;
    message: string;
    show: boolean;
}
