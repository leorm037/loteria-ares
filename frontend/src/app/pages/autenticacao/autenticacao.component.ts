import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
import { Router } from '@angular/router';
import { AutenticacaoService } from '../../services/autenticacao.service';

@Component({
  selector: 'app-autenticacao',
  imports: [ReactiveFormsModule],
  templateUrl: './autenticacao.component.html'
})
export class AutenticacaoComponent implements OnInit {

  autenticacaoForm!: FormGroup;

  constructor(
    private formBuilder: FormBuilder,
    private service: AutenticacaoService,
    private router: Router
  ) { }

  ngOnInit(): void {
    this.autenticacaoForm = this.formBuilder.group({
      email: [null, [Validators.required, Validators.email]],
      senha: [null, Validators.required]
    });
  }

  public autenticar() {
    if (this.autenticacaoForm.valid) {
      const email = this.autenticacaoForm.value.email;
      const senha = this.autenticacaoForm.value.senha;

      this.service.autenticacao(email,senha).subscribe({
        next: data => console.log(data),
        error: error => console.error(error)
      });

      console.log("formulario valido", this.autenticacaoForm.value);
    } else {
      console.error("formulario invalido", this.autenticacaoForm.value);
    }

    this.router.navigateByUrl('/');
  }
}
