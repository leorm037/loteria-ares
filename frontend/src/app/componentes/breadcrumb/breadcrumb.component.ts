import { Component, OnInit } from '@angular/core';
import { Breadcrumb } from '../../interfaces/breadcrumb';
import { BreadcrumbService } from '../../services/breadcrumb.service';
import { RouterModule } from '@angular/router';

@Component({
  selector: 'app-breadcrumb',
  imports: [RouterModule],
  templateUrl: './breadcrumb.component.html',
  styleUrl: './breadcrumb.component.css'
})
export class BreadcrumbComponent implements OnInit {

  public breadcrumbs!: Breadcrumb[] | null;

  constructor(
    private service: BreadcrumbService
  ) { }

  ngOnInit(): void {
      this.service.getBreadcrumbs().subscribe(
        breadcrumbs => {
          this.breadcrumbs = breadcrumbs;
        }
      );
  }


}
