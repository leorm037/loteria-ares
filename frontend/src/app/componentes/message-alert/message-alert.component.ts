import { Component, OnInit } from '@angular/core';
import { MessageAlert } from '../../interfaces/message-alert';
import { MessageAlertService } from '../../services/message-alert.service';

@Component({
  selector: 'app-message-alert',
  imports: [],
  templateUrl: './message-alert.component.html'
})
export class MessageAlertComponent implements OnInit {

  public messagesAlert!: MessageAlert[] | null;

  constructor(
    private service: MessageAlertService
  ) { }

  ngOnInit(): void {
    this.service.getMessagesAlert()
      .subscribe(
        messagesAlert => {
          this.messagesAlert = messagesAlert;
        }
      );
  }

}
