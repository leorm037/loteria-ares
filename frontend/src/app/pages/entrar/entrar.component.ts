import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
import { Router, RouterModule } from '@angular/router';
import { AutenticacaoService } from '../../services/autenticacao.service';
import { Breadcrumb } from '../../interfaces/breadcrumb';
import { BreadcrumbService } from '../../services/breadcrumb.service';

@Component({
  selector: 'app-entrar',
  imports: [ReactiveFormsModule, RouterModule],
  templateUrl: './entrar.component.html'
})
export class EntrarComponent implements OnInit {

  public autenticacaoForm!: FormGroup;

  private readonly breadcrumbs: Breadcrumb[] = [
    {
      iconClass: "bi bi-door-closed",
      texto: "Entrar"
    }
  ];

  constructor(
    private formBuilder: FormBuilder,
    private service: AutenticacaoService,
    private router: Router,
    private breadcrumbService: BreadcrumbService
  ) { }

  ngOnInit(): void {
    this.autenticacaoForm = this.formBuilder.group({
      email: [null, [Validators.required, Validators.email]],
      senha: [null, Validators.required],
      lembrar: [false]
    });

    this.breadcrumbService.sendBreadcrumb(this.breadcrumbs);
  }

  public autenticar() {
    if (this.autenticacaoForm.valid) {
      const email = this.autenticacaoForm.value.email;
      const senha = this.autenticacaoForm.value.senha;

      this.service.autenticacao(email, senha).subscribe({
        next: data => console.log(data),
        error: error => console.error(error)
      });

      console.log("formulario valido", this.autenticacaoForm.value);
    } else {
      console.error("formulario invalido", this.autenticacaoForm.value);
    }

    this.router.navigateByUrl('/');
  }
}
