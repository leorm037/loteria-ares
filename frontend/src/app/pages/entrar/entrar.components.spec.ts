import { ComponentFixture, TestBed } from "@angular/core/testing";
import { EntrarComponent } from "./entrar.component";
import { FormBuilder, ReactiveFormsModule } from "@angular/forms";
import { HttpClientTestingModule } from "@angular/common/http/testing";
import { RouterTestingModule } from "@angular/router/testing";
import { AutenticacaoService } from "../../services/autenticacao.service";
import { BreadcrumbService } from "../../services/breadcrumb.service";
import { MessageAlertService } from "../../services/message-alert.service";

describe('EntrarComponent', () => {
    let breadcrumbService: BreadcrumbService;
    let component: EntrarComponent;
    let fixture: ComponentFixture<EntrarComponent>;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                ReactiveFormsModule,
                HttpClientTestingModule,
                RouterTestingModule
            ],
            providers: [
                FormBuilder,
                AutenticacaoService,
                BreadcrumbService,
                MessageAlertService
            ]
        });
        breadcrumbService = TestBed.inject(BreadcrumbService);
        fixture = TestBed.createComponent(EntrarComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
    });

    it('Deve ser criado.', () => {
        expect(component).toBeTruthy();
    });

    it('Deve ser criado o formulário de autenticação.', () => {
        expect(component.autenticacaoForm.value).toEqual({
            email: null,
            senha: null,
            lembrar: false
        });
    });

    it('Deve ter o breadcrumb definido.', () => {
        expect(breadcrumbService.getBreadcrumbs()).toMatchSnapshot();
    });
});