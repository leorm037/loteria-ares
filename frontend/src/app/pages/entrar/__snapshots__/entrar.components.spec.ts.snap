// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntrarComponent Deve ter o breadcrumb definido. 1`] = `
Observable {
  "source": BehaviorSubject {
    "_value": [
      {
        "iconClass": "bi bi-door-closed",
        "texto": "Entrar",
      },
    ],
    "closed": false,
    "currentObservers": [],
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
}
`;
